src/mongocxx/collection.cpp

```diff
@@ -426,12 +426,89 @@ std::int64_t collection::count(bsoncxx::document::view filter, const options::co
 }
 
 bsoncxx::document::value collection::create_index(bsoncxx::document::view keys,
-                                                  bsoncxx::document::view options) {
+                                                  const options::index& options) {
     scoped_bson_t bson_keys{keys};
+    scoped_bson_t weights{options.weights()};
+    scoped_bson_t partial_filter_expression{options.partial_filter_expression()};
     bson_error_t error;
+    ::mongoc_index_opt_geo_t geo_opt{};
+    ::mongoc_index_opt_t opt{};
+    ::mongoc_index_opt_wt_t wt_opt{};
+    libmongoc::index_opt_init(&opt);
+
+    if (options.background()) {
```

Comment:
It could be a method that took a void * and populated it instead:
```
void export_options(void* opts) {
   // cast to mongoc_index_opt_t *
   // carry on
}
```

This is also sketchy but is marginally less sketchy, I think